library/ssl_tls13_client.c

```diff
@@ -1335,6 +1335,53 @@ static int ssl_tls13_is_downgrade_negotiation( mbedtls_ssl_context *ssl,
     return( 0 );
 }
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
```

Comment:
The early data indication extension is very similar in NewSessionTicket, ClientHello and EncryptedExtensions thus in that case it would be good to have only one parsing function for the three cases. The function should just do the parsing though thus all the code related to `early_data_status` currently in this function should be moved to `ssl_tls13_parse_encrypted_extensions()`.